File: iOS/Code/Swift/bugTrap/bugTrapKit/Controllers/BtViewController.swift
//
//  BtViewController.swift
//  bugTrap
//
//  Created by Colby L Williams on 11/20/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import UIKit

class BtViewController: UIViewController {

	var screenName: String? = nil
	
	required init(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override func viewDidAppear(animated: Bool) {
		super.viewDidAppear(animated)
		
		updateScreenName()
	}
	
	func updateScreenName () {
		if let name = screenName {
			Analytics.Shared.screenView(name)
		}
	}
	
	func updateScreenName (name: String) {
		if name != screenName {
			screenName = name
			updateScreenName()
		}
	}
	
	override func prefersStatusBarHidden() -> Bool {
		return true
	}
	override func preferredStatusBarStyle() -> UIStatusBarStyle {
		return .LightContent
	}
}

File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Controllers/BtViewController.cs
using System;
using UIKit;

namespace bugTrapKit
{
	public class BtViewController : UIViewController
	{

		public BtViewController ()
		{
		}

		public BtViewController (IntPtr handle) : base(handle)
		{
			
		}

		public string ScreenName { get; set; }

		public override void ViewDidAppear (bool animated)
		{
			base.ViewDidAppear(animated);

			UpdateScreenName();
		}

		public void UpdateScreenName ()
		{
			if (!string.IsNullOrWhiteSpace(ScreenName)) {
				// Analytics.Shared.ScreenView(screenName);
			}
		}

		public void UpdateScreenName (string name)
		{
			if (name != ScreenName) {
				ScreenName = name;
				UpdateScreenName();
			}
		}

		public override bool PrefersStatusBarHidden () => true;

		public override UIStatusBarStyle PreferredStatusBarStyle () => UIStatusBarStyle.LightContent;
	}
}